main.go

```diff
@@ -23,219 +21,69 @@ import (
 	}
 
-	//This is very static, should be changed to above
-	gasLimit = gasLimit * 10
-
-	// Fill the transactions, maybe sign them and then push?
-	for index := 0; index < stdInt; index++ {
-		// Make random bytes to differ tx (May not work as expected)
-		//token := make([]byte, 16)
-		//rand.Read(token)
-		addrToSend := AddressToSend
-		currentTx := types.NewTransaction(nonce, addrToSend, amount, gasLimit, gasPrice, make([]byte, 0))
-		signedTx, err := types.SignTx(currentTx, types.NewEIP155Signer(ChainId), privateKey)
-
-		if index%10 == 0 {
-			fmt.Printf("\n Signed new tx, %d", index)
-		}
-
-		if nil != err {
-			err = fmt.Errorf("\n error occured at txId: %d of total: %d, err: %s", index, stdInt, err.Error())
-
-			return err, transactions
-		}
-
-		transactions = append(transactions, signedTx)
+	_ = drill.New(ethCli, cfg.privateKey, cfg.addressToSend, cfg.chainID)
 
-		// Nonce get call is done only once before the loop, may lead to problems
-		nonce++
-	}
-
-	return
+	fmt.Printf("\n Running chaindriller on IPC: %s", cfg.ipcEndpoint)
 }
 
-func SendBulkOfSignedTransaction(
-	client *ethclient.Client,
-	transactions []*types.Transaction,
-) (err error, finalReport FinalReport) {
-	ctx := context.Background()
-	finalReport.Transactions = transactions
-	finalReport.Errors = make([]error, 0)
-	finalReport.TransactionHashes = make([]string, 0)
-
-	var (
-		waitGroup         sync.WaitGroup
-		routinesWaitGroup sync.WaitGroup
-	)
-
-	//Lets make some sense in possible routines at once with the lock. I suggest max 1k
-	minRoutinesUp := len(transactions)
-
-	if minRoutinesUp > 5000 {
-		minRoutinesUp = 5000
+func getConfig() (cfg Config) {
+	cfg.ipcEndpoint = os.Getenv("IPC_ENDPOINT")
+	if cfg.ipcEndpoint == "" {
+		cfg.ipcEndpoint = defaultIPCEndpoint
 	}
 
-	routinesWaitGroup.Add(minRoutinesUp)
-
-	for index, transaction := range transactions {
-		waitGroup.Add(1)
-
-		if index%100 == 0 {
-			fmt.Printf("\nStarting routine index: %d", index)
-		}
-
-		go func(transaction *types.Transaction, index int) {
-			routinesWaitGroup.Done()
-			routinesWaitGroup.Wait()
-
-			if index%1000 == 0 {
-				fmt.Printf("\nStarting routines : %d", index)
-			}
-
-			err = client.SendTransaction(ctx, transaction)
-			transactionHash := transaction.Hash()
-
-			if nil != err {
-				finalReport.Errors = append(finalReport.Errors, err)
-			}
-
-			finalReport.TransactionHashes = append(finalReport.TransactionHashes, transactionHash.String())
-			waitGroup.Done()
-		}(transaction, index)
-	}
-
-	waitGroup.Wait()
-
-	return
-}
+	chainIDstr := os.Getenv("CHAIN_ID")
+	chainID, err := strconv.ParseInt(chainIDstr, 10, 64)
 
-// newClient creates a client with specified remote URL.
-func newClient(ipcEndpoint string) *ethclient.Client {
-	client, err := ethclient.Dial(ipcEndpoint)
-	if err != nil {
-		panic(fmt.Sprintf("Could not connect to ethereum node url: %s, Err: %s", ipcEndpoint, err.Error()))
+	if err == nil {
+		cfg.chainID = big.NewInt(chainID)
+	} else {
```

Comment:
please never use else, use early statements or defaults

Reply:
Doesn't really apply here, because there are calculations in both branches. 
In one of them we construct `big.Int` and in the other we `Printf` info to console.